### ISSUE TITLE
bizarre amount of newlines in generated templates

### ISSUE THREAD

**Issue by cartazio** [simonmar/alex#105]
[AlexWrapper-basic copy.txt](https://github.com/simonmar/alex/files/1003043/AlexWrapper-basic.copy.txt)

theres thousands of newlines in that attached file!
also some of the source / line attributions seem likely to cause nondeterminstic builds under some conditions
eg 
```

{-# LINE 1 "/var/folders/py/wgp_hj9d2rl3cx48yym_ynj00000gn/T/ghc64947_0/ghc_2.h" #-}
```

cc @hvr

**Comment by erikd** [comment]
I agree that it would probably be nice to remove excessive newlines, but I can't see how these, or the source line attributions could cause non-deterministic builds. Can you generate a test case that show this?

**Comment by erikd** [comment]
Ok so the non-determinacy comes from things like:
```
{-# LINE 1 "/var/folders/py/wgp_hj9d2rl3cx48yym_ynj00000gn/T/ghc64947_0/ghc_2.h" #-}
```
but does that come from Alex or GHC ?

**Comment by hvr** [comment]
That looks a lot like a tmp-file generated by GHC... I wonder if GHC should #line-annotate positions in ephemeral tmp files... (especially if -keep-tmp-files is not active)

**Comment by andreasabel** [comment]
Is this still a problem?  Cannot reproduce atm.  
We need a reproducer to report this upstream.
